File: median.js
function intcmp(a,b){
   return a-b;
}

function middle(a){
   var sorted=a.sort(intcmp);

   console.log(a)
   return sorted[a.length>>1];
}

function even_integer(val){
   if(val<0){
      return false;
   }

   if(val!=(val|0)){
      return false;
   }

   if(val==((val>>1)<<1)){
      return false;
   }

   return true;
}

function bucket(samples,gap,fun){
   var i=0;
   var l=[];
   var r=[];
   var half_gap=0;
   var extended;
   var ret=[];

   if(!even_integer(gap)){
      return null;
   }

   half_gap=(gap/2)|0;

   for(i=0;i<half_gap;++i){
      l.push(samples[0])
   }

   for(i=0;i<half_gap;++i){
      r.push(samples[samples.length-1])
   }

   extended=l.concat(samples,r);
   
   for(i=0;i<samples.length;++i){
      ret.push(fun(extended.slice(i,i+gap)))
   }

   console.log(ret)
}

bucket([1,2,3,4,5,6],5,middle);

phantom.exit();

File: median.py
def med(a):
    return sorted(a)[len(a)/2]

def bucket(samples,gap,fun):
    if not gap%2:
        sys.exit()
    extended=[samples[0]]*(gap/2)+samples+[samples[-1]]*(gap/2)
    ret=[]
    for i in range(0,len(samples)):
        ret.append(fun(extended[(i):(i+gap)]))
    return ret
   
if __name__ == "__main__":
    print bucket([1,2,9,4,5,6],5,med)
